src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.service;
+
+import core.basesyntax.operation.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.transaction.FruitTransaction;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+    private Map<String, Integer> fruits = new HashMap<>();
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void addFruits(String fruit, int quantity) {
+        fruits.put(fruit, fruits.getOrDefault(fruit, 0) + quantity);
+    }
+
+    @Override
+    public int getQuantity(String fruit) {
+        return fruits.getOrDefault(fruit, 0);
+    }
+
+    @Override
+    public Map<String, Integer> getFruits() {
+        return fruits;

```

Comment:
Returning the actual `fruits` map from the `getFruits()` method can lead to the map being modified outside of this class, which can lead to bugs and make the code harder to debug. Consider returning a copy of the map or an unmodifiable view of the map instead. You can use `Collections.unmodifiableMap(fruits)` to get an unmodifiable view of the map.